src/fromager/external_commands.py

```diff
@@ -2,15 +2,16 @@
 import os
 import shlex
 import subprocess
+import typing
 
 logger = logging.getLogger(__name__)
 
 
 # based on pyproject_hooks/_impl.py: quiet_subprocess_runner
 def run(
-    cmd: list[str],
+    cmd: typing.Sequence[str],
     cwd: str | None = None,
-    extra_environ: dict | None = None,
+    extra_environ: dict[str, typing.Any] | None = None,
```

Comment:
That should be also `dict[str, str]`
```suggestion
    extra_environ: dict[str, str] | None = None,
```

Reply:
I have added a TODO item to the ticket.